File: src/main/java/link/infra/packwiz/installer/metadata/hash/HashUtils.java
package link.infra.packwiz.installer.metadata.hash;

import java.util.HashMap;
import java.util.Map;

public class HashUtils {
	private static final Map<String, IHasher> hashTypeConversion = new HashMap<String, IHasher>();
	static {
		hashTypeConversion.put("sha256", new HashingSourceHasher("sha256"));
		hashTypeConversion.put("murmur2", new Murmur2Hasher());
	}

	public static IHasher getHasher(String type) throws Exception {
		IHasher hasher = hashTypeConversion.get(type);
		if (hasher == null) {
			throw new Exception("Hash type not supported: " + type);
		}
		return hasher;
	}

	public static Object getHash(String type, String value) throws Exception {
		if (hashTypeConversion.containsKey(type)) {
			return hashTypeConversion.get(type).getHash(value);
		}

		throw new Exception("Hash type not supported: " + type);
	}

}

File: src/main/kotlin/link/infra/packwiz/installer/metadata/hash/HashUtils.kt
package link.infra.packwiz.installer.metadata.hash

object HashUtils {
	private val hashTypeConversion: Map<String, IHasher> = mapOf(
			"sha256" to HashingSourceHasher("sha256"),
			"sha512" to HashingSourceHasher("sha512"),
			"murmur2" to Murmur2Hasher(),
			"sha1" to HashingSourceHasher("sha1")
	)

	@JvmStatic
	@Throws(Exception::class)
	fun getHasher(type: String): IHasher {
		return hashTypeConversion[type] ?: throw Exception("Hash type not supported: $type")
	}

	@JvmStatic
	@Throws(Exception::class)
	fun getHash(type: String, value: String): Hash {
		return hashTypeConversion[type]?.getHash(value) ?: throw Exception("Hash type not supported: $type")
	}
}